packages/core/content-manager/server/src/services/document-metadata.ts

```diff
@@ -233,8 +290,43 @@ export default ({ strapi }: { strapi: Core.Strapi }) => ({
     }
 
     const meta = await this.getMetadata(uid, document, opts);
+    let { availableLocales, availableStatus } = meta;
+
+    // Always sanitize passwords
+    const defaultSanitizer = sanitize.sanitizers.sanitizePasswords({
+      schema: strapi.getModel(uid),
+      getModel(uid: string) {
+        return strapi.getModel(uid as UID.Schema);
+      },
+    });
```

Comment:
could we create the sanitizer only once, so there is no need to duplicated the availableLocales & status code?

Also, shouldn't it be more adecuate to sanitize data on the controller level? even if it's duplicated, we do the same for the document-manager service.

Good catch on sanitizing those fields!

Reply:
Which leds me to think, what happens if a user doesn't have read permission of a field required to check the publishing validations?